### PR TITLE
Add NativeUpdateHistory to ProtocolSettings

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by Qiao-Jin** [comment]
Is updating history logic still under development? It seems only deploying index is used.

**Comment by erikzhang** [comment]
Currently, it is enough for us to have the update history only. Updating native contracts needs to update the core, so we can change the logic at that time.

File: src/neo/SmartContract/ApplicationEngine.Contract.cs [hunk 1]
```diff
@@ -88,17 +88,23 @@ protected internal void NativeOnPersist()
             if (Trigger != TriggerType.OnPersist)
                 throw new InvalidOperationException();
             foreach (NativeContract contract in NativeContract.Contracts)
-                if (ProtocolSettings.GetNativeActivation(contract.Name) <= PersistingBlock.Index)
+            {
+                uint activeIndex = ProtocolSettings.NativeUpdateHistory[contract.Name][0];
```
[the remark this answers is not in the document]
Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Contract.cs:92]: Now the value is an array, we don't have a default value. So we have to set for all the native contracts.

File: src/neo/SmartContract/Native/NativeContract.cs [hunk 2]
```diff
@@ -103,7 +103,8 @@ public static NativeContract GetContract(UInt160 hash)
 
         internal void Invoke(ApplicationEngine engine, byte version)
         {
-            if (engine.ProtocolSettings.GetNativeActivation(Name) > Ledger.CurrentIndex(engine.Snapshot))
+            uint activeIndex = engine.ProtocolSettings.NativeUpdateHistory[Name][0];
```
[the remark this answers is not in the document]
Comment (erikzhang) [reply src/neo/SmartContract/Native/NativeContract.cs:106]: In verification trigger, it is possible.

[the remark this answers is not in the document]
Comment (erikzhang) [reply src/neo/SmartContract/Native/NativeContract.cs:106]: But I think it's better to move it to `System.Contract.CallNative`.

**Review by shargon — APPROVED** [review]

**Pull request merged**